lib/vnet/daemon/client_darwin.go

```diff
@@ -0,0 +1,74 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+//go:build darwin
+// +build darwin
+
+package daemon
+
+// #cgo CFLAGS: -Wall -xobjective-c -fblocks -fobjc-arc -mmacosx-version-min=10.15
+// #cgo LDFLAGS: -framework Foundation -framework ServiceManagement
+// #include "client_darwin.h"
+import "C"
```

Comment:
Although within this PR the `daemon` package is only ever called from code compiled on darwin only, in the future PRs I'm going to add some common Go code that defines structs used on other platforms as well (it's used only in type signatures there).

Hence why every file, both Go and Obj-C, has `_darwin` in it. cgo on Linux doesn't let you build a package if it includes Obj-C files.

Reply:
I added build tags to the .m files on touchid (and similar packages). That works too.

- https://github.com/gravitational/teleport/blob/7310b2f17ac0fe643fb1552ba2e06b395090a10e/lib/auth/touchid/authenticate.m#L1-L2